scammer-label-propagation/src/agent.py

```diff
@@ -0,0 +1,149 @@
+from concurrent.futures import ThreadPoolExecutor
+
+from src.main import run_all
+from src.constants import attacker_bots, ATTACKER_CONFIDENCE, N_WORKERS, CHAIN_ID, MAX_FINDINGS
+
+logging.basicConfig(filename=f"logs.log", level=logging.INFO, 
+                    format='%(levelname)s:%(asctime)s:%(name)s:%(lineno)d:%(message)s')
+logger = logging.getLogger(__name__)
+
+
+def run_all_extended(central_node, alert_id):
+    try:
+        attackers_df = run_all(central_node)
+    except Exception as e:
+        logger.error(f"{central_node}:\tError running run_all in a thread: {e}")
+        return []
+    # Now we put things into a list of findings
+    all_findings_list = []
+    finding_dict = {
+            'name': 'scammer-label-propagation',
+            'description': 'Address marked as scammer by label propagation',
+            'alert_id': 'SCAMMER-LABEL-PROPAGATION-1',
+            'severity': forta_agent.FindingSeverity.High,
+            'type': forta_agent.FindingType.Scam
+        }
+    if attackers_df.shape[0] > MAX_FINDINGS:
+        logger.error(f"{central_node}:\tToo many attackers found: {attackers_df.shape[0]}. Labels may be compromised")
+    else:
+        for row_idx in range(attackers_df.shape[0]):
+            attacker_info = attackers_df.iloc[row_idx]
+            logger.info(f'{central_node}:\tNew attacker info: {attacker_info}')
+            label_dict = {
+                'entity': attacker_info.name,
+                'label': 'scammer-eoa',
+                'confidence': attacker_info['n_predicted_attacker']/10 * attacker_info['mean_probs_attacker'],
+                'entity_type': forta_agent.EntityType.Address,
+                'metadata': {
+                    'central_node': central_node,
+                    'central_node_alert_id': alert_id,
+                }
+            }
+            finding_dict['labels'] = [forta_agent.Label(label_dict)]
+            all_findings_list.append(Finding(finding_dict))
+    return all_findings_list
+        
+        
+def initialize():
+    global executor
+    executor = ThreadPoolExecutor(max_workers=N_WORKERS)
+    global addresses_analyzed
+    addresses_analyzed = []
+    global global_futures
+    global_futures = {}
+    global global_alerts
+    global_alerts = []
+
+    subscription_json = []
+    for bot in attacker_bots:
+        subscription_json.append({"botId": bot, "chainId": CHAIN_ID})
+    alert_config = {"alertConfig": {"subscriptions": subscription_json}}
+    logger.info(f"Initializing scammer label propagation bot. Subscribed to bots successfully: {alert_config}")
+    return alert_config
+
+
+def provide_handle_alert(w3):
+    logger.debug("provide_handle_alert called")
+    
+
+    def handle_alert(alert_event: forta_agent.alert_event.AlertEvent) -> list:
+        logger.debug("handle_alert inner called")
+        logger.debug(f"AlertId:{alert_event.alert_id};\tName:{alert_event.name};\tAlertHash:{alert_event.hash}")
```

Comment:
you may want to add botID to logging as well

Reply:
Added